instance/tests/integration/integration_instance.py

```diff
@@ -71,6 +71,35 @@ def assert_instance_up(self, instance):
         for url in [instance.url, instance.lms_preview_url, instance.studio_url]:
             check_url_accessible(url)
 
+    def assert_appserver_firewalled(self, instance):
+        """
+        Ensure the instance's appserver is not exposing any services it shouldn't be
+        """
+        instance.refresh_from_db()
+        self.assertIsNotNone(instance.active_appserver)
+        server_ip = instance.active_appserver.server.public_ip
+        ports_should_be_open = [22, 80]  # 443 may or may not be open, depending on instance.protocol
```

Comment:
I believe that right now, if you set "protocol=https" on an Instance, then its appservers will be listening on both port 80 and 443. If you leave the default "protocol=http", then the appservers only listen on port 80. It's based on the ["NGINX_ENABLE_SSL" nginx setting](https://github.com/open-craft/opencraft/blob/0117950b766481f120e663f572ef22824f2f6a2d/instance/templates/instance/ansible/vars.yml#L30). Note that even if the AppServer is listening on port 443, it doesn't have a valid SSL certificate.